bot/common.py

```diff
@@ -173,6 +178,22 @@ def check_url_needs_to_be_skipped(wikilink, page):
     full_url = page.full_url()
     check_has_fragment(full_url)
     if page.isRedirectPage():
+        try:
+            wp.ItemPage.fromPage(page)
+        except wp.NoPage:
+            # Page is a redirect without its own wikidata item -
+            # everything's OK, we can safely fix the redirect.
+            # Examples of this are
+            # https://en.wikipedia.org/w/index.php?title=Star_Wars_Episode_I:_The_Phantom_Menace_(soundtrack)&redirect=no
+            pass
+        else:
+            # Page is a redirect, but has its own wikidata item - it's better
+            # to not touch this.
+            # Examples of this are
+            # https://en.wikipedia.org/wiki/Hybrid_Theory_%28EP%29, which is a
+            # redirect on en.wp, but has its own article on lots of other
+            # wikipedias and therefore its own wikidata item.
```

Comment:
If possible, if the Wikipedia page matches a Wikidata item already present on the entity, we could just remove the unneeded Wikipedia page, too.

Reply:
Unfortunately, the `musicbrainz_bot` module used for editing only supports adding or editing URLs, but not removing them :-(